File: backend/ElectronicService/src/ElectronicService.Domain/Catalog/ImportBatches/CatalogImportRowStatus.cs
namespace ElectronicService.Domain.Catalog.ImportBatches;

public enum CatalogImportRowStatus
{
    None = 0,

    /*
     * Полная проверка невозможна,
     * пока не закончено сопоставление колонок.
     */
    PendingMapping = 1,

    /*
     * Строка прошла текущую валидацию.
     */
    Valid = 2,

    /*
     * В строке есть ошибки, которые
     * необходимо исправить на сайте.
     */
    Error = 3
}

File: frontend/electronic-service-web/src/features/catalogImports/model/catalogImportRowStatus.ts
import type {
  CatalogImportRowFilterStatus,
  CatalogImportRowStatus,
} from "./types";

const statusLabels: Record<CatalogImportRowStatus, string> = {
  None: "Не определён",
  PendingMapping: "Ожидает сопоставления",
  Valid: "Корректная",
  Error: "С ошибками",
};

export function getCatalogImportRowStatusLabel(
  status: CatalogImportRowStatus,
): string {
  return statusLabels[status];
}

export function getCatalogImportRowFilterStatusLabel(
  status: CatalogImportRowFilterStatus,
): string {
  return statusLabels[status];
}